src/gui_utils.py

```diff
@@ -219,23 +219,27 @@ def createStlActorInOrigin(filename, colorize=False):
 
     if colorize:
         actor = ColorizedStlActor(output)
+        actor = setTransformFromSettings(actor)
     else:
         actor = StlActor(output)
+        transform = vtk.vtkTransform()
+        actor.SetUserTransform(transform)
 
+    return actor
+
+def setTransformFromSettings(actor):
     s = sett()
     transform = vtk.vtkTransform()
-    transform.Translate(s.hardware.plane_center_x, s.hardware.plane_center_y, s.hardware.plane_center_z)
-    transform.Translate(s.slicing.originx, s.slicing.originy, s.slicing.originz)
 
-    transform.RotateZ(s.slicing.rotationz)
-    transform.RotateX(s.slicing.rotationx)
-    transform.RotateY(s.slicing.rotationy)
-
-    transform.Scale(s.slicing.scalex, s.slicing.scaley, s.slicing.scalez)
+    m = vtkMatrix4x4()
+    for i in range(3):
```

Comment:
3? please add comments everywhere. 
And maybe better to have separate functions that create matrices from data